packages/auth0_api_python/tests/test_api_client.py

```diff
@@ -388,3 +404,1153 @@ async def test_verify_access_token_fail_no_audience_config():
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(InvalidDpopProofError) as err:
+        await api_client.verify_dpop_proof(
+            access_token=access_token,
+            proof=dpop_proof,
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "unexpected jwt 'typ'" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_invalid_alg():
+    """
+    Test that a DPoP proof with unsupported algorithm fails verification.
+    """
+    access_token = "test_token"
+
+    valid_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    parts = valid_proof.split('.')
+    header = json.loads(base64.urlsafe_b64decode(parts[0] + '==').decode('utf-8'))
+    header['alg'] = 'RS256'  # Invalid algorithm for DPoP (should be ES256)
+
+    modified_header = base64.urlsafe_b64encode(
+        json.dumps(header, separators=(',', ':')).encode('utf-8')
+    ).decode('utf-8').rstrip('=')
+
+    invalid_proof = f"{modified_header}.{parts[1]}.{parts[2]}"
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(InvalidDpopProofError) as err:
+        await api_client.verify_dpop_proof(
+            access_token=access_token,
+            proof=invalid_proof,
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "unsupported alg" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_no_jwk():
+    """
+    Test that a DPoP proof missing 'jwk' header fails verification.
+    """
+    access_token = "test_token"
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/resource",
+        header_overrides={"jwk": None}  # Remove jwk header
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(InvalidDpopProofError) as err:
+        await api_client.verify_dpop_proof(
+            access_token=access_token,
+            proof=dpop_proof,
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "missing or invalid jwk" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_invalid_jwk_format():
+    """
+    Test that a DPoP proof with invalid 'jwk' format fails verification.
+    """
+    access_token = "test_token"
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/resource",
+        header_overrides={"jwk": "invalid_jwk"}  # Invalid jwk format
```

Comment:
This seems to be testing the same thing as the test below. The invalid JWK used here should be invalid for a different reason than having a `d` parameter.

Reply:
My bad, I was confused by `invalid_jwk` having the same name.